pkg/resources/secret_with_basic_authentication.go

```diff
@@ -107,54 +133,55 @@ func ReadContextSecretWithBasicAuthentication(ctx context.Context, d *schema.Res
 	return nil
 }
 
-func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func UpdateContextSecretWithBasicAuthentication(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
 	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	if request := handleSecretUpdate(id, d); request != nil {
-		if err := client.Secrets.Alter(ctx, request); err != nil {
-			return diag.FromErr(err)
-		}
+	set := &sdk.SecretSetRequest{}
+	unset := &sdk.SecretUnsetRequest{}
+	handleSecretUpdate(d, set, unset)
+	setForFlow := &sdk.SetForFlowRequest{
+		SetForBasicAuthentication: &sdk.SetForBasicAuthenticationRequest{},
 	}
 
 	if d.HasChange("username") {
 		username := d.Get("username").(string)
-
-		request := sdk.NewAlterSecretRequest(id)
-		setRequest := sdk.NewSetForBasicAuthenticationRequest().WithUsername(username)
-		request.WithSet(*sdk.NewSecretSetRequest().WithSetForBasicAuthentication(*setRequest))
-
-		if err := client.Secrets.Alter(ctx, request); err != nil {
-			return diag.FromErr(err)
-		}
+		setForFlow.SetForBasicAuthentication.WithUsername(username)
+		set.WithSetForFlow(*setForFlow)
```

Comment:
Im not sure this will work properly, if I change `set` after all HasChanged blocks, and none of them will actually change anything to the `setForFlow` struct, and then use `set.WithSetForFlow(*setForFlow)` just before the reflect   this will execute Alter WithSet because set will have `setForFlow` (although it does not change anything) and result in SQL error since this Alter will not change anything

I think that set should be changed only when the value has been actually changed.
The other way i could think of is to do additional check after all HasChanged blocks that will check if setForFlow has been modified

Reply:
At first, I also did not like modifying the set here, but @sfc-gh-fbudzynski is right, that we would have to check first if setForFlow has changed at all before adding it to set. I am open to both options TBH.

Reply:
In the end I have changed it to what @sfc-gh-jmichalak suggested, ok we do need to check it but the set itself is not being modified multiple times